File: Back/Src/ProEventos.Domain/Models/Evento.cs
using System;
using System.Collections.Generic;

namespace ProEventos.Domain.Models
{
    public class Evento
    {
        public int Id { get; set; }
        public required string Local { get; set; }
        public required DateTime DataEvento { get; set; }
        public required string Tema { get; set; }
        public required int QtdPessoas { get; set; }
        public required string ImagemUrl { get; set; }
        public string Telefone { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public ICollection<Lote> Lotes { get; set; } = new List<Lote>();
        public ICollection<RedeSocial>? RedesSociais { get; set; }
        public ICollection<PalestranteEvento>? PalestrantesEventos { get; set; }
    }
}

File: Front/ProEventos-app/src/app/models/Evento.ts
import { Palestrante } from "./palestrante";
import { RedeSocial } from "./rede-social";
import { Lote } from "./lote";

export interface Evento {
  id: number;
  tema: string;
  descricao: string;
  local: string;
  dataEvento: string;      
  dataEventoDate?: Date;   
  horario: string;
  qtdPessoas: number;
  maxPessoas: number;
  categoria: string;
  imagemUrl: string;
  telefone: string;
  email: string;
  visualizacoes: number;
  TodaDescricao: string;
  lotes: Lote[];
  redesSociais?: RedeSocial[];
  palestrantesEventos?: Palestrante[];
}

